From 7e5b312ffc741a348e85b696a228e5bf3b25ad2e Mon Sep 17 00:00:00 2001
From: Tobias Jakobi <tjakobi@math.uni-bielefeld.de>
Date: Thu, 6 Jun 2024 23:17:00 +0200
Subject: [PATCH] hwdb: add keyboard mappings for the Ayaneo Kun face buttons

See comment in hwdb file for details.

(cherry picked from commit 9afc6ac87c01c70917e996e7d3e2d381bf2b4f07)

Related: RHEL-5950
---
 hwdb.d/60-keyboard.hwdb | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/hwdb.d/60-keyboard.hwdb b/hwdb.d/60-keyboard.hwdb
index 461b9a7cb4..f1ea0f58b1 100644
--- a/hwdb.d/60-keyboard.hwdb
+++ b/hwdb.d/60-keyboard.hwdb
@@ -279,6 +279,22 @@ evdev:name:gpio-keys:phys:gpio-keys/input0:ev:3:dmi:*:svnASUSTeKCOMPUTERINC.:pnT
 evdev:name:gpio-keys:phys:gpio-keys/input0:ev:100003:dmi:*:svnASUSTeKCOMPUTERINC.:pnTF103C*:*
  KEYBOARD_KEY_0=volumedown
 
+###########################################################
+# Ayaneo
+###########################################################
+
+# The Ayaneo Kun is a portable PC handheld console. Four of its
+# face buttons are not exposed through the integrated Xbox360
+# controller, but by a raw AT keyboard.
+#
+# All four of these buttons generate a multi-scancode sequence
+# consisting of Left_Meta, Right_Ctrl and another scancode.
+evdev:atkbd:dmi:bvn*:bvr*:bd*:svnAYANEO:pnKUN:pvr*
+ KEYBOARD_KEY_18=f13                              # LC (next to the left shoulder button)
+ KEYBOARD_KEY_20=f14                              # RC (next to the right shoulder button)
+ KEYBOARD_KEY_30=f15                              # Start (upper-right corner of screen)
+ KEYBOARD_KEY_28=f16                              # Ayaneo (lower-right corner of screen)
+
 ###########################################################
 # BenQ
 ###########################################################
